### PR TITLE
MultiPartBodyParser Refactor

### PR DESCRIPTION
**Pull request by dsperling**
## Description
The refactor centers around the addition of an NSData extension function "components".  It operates similar to String.components where NSData can be split into an array of NSData elements using a separator.  In the case of MultiPart parsing, the separator is the boundary.

String components function:

      func components(separatedBy separator: String) -> [String] { ... }

New NSData components extension:

      func components(separatedBy separator: NSData) -> [NSData] { ... }

The new components function may be useful in other parts of Kiitura.

## Motivation and Context
During development of MultiPartBodyParser.swift, there was a goal of an eventual refactor.  No functionality was changed.

## How Has This Been Tested?
Tested on Mac and Linux.  A new negative test case was added as well.

## Checklist:

- [x] I have submitted a [CLA form](https://github.com/IBM-Swift/CLA)
- [ ] If applicable, I have updated the documentation accordingly.
- [x] If applicable, I have added tests to cover my changes.

### CODE REVIEW

**Comment by dsperling** [comment]
@dfirsht - Here is the refactor we spoke about for issue #438.  There was an initial Linux build failure which I fixed, however the last Linux build appears to have stalled (72ed16d).  Is there a way to attempt the Linux build again?

**Comment by dfirsht** [comment]
@dsperling I restarted it.

**Comment by dsperling** [comment]
@dfirsht thanks.  That is more like it.

File: Sources/Kitura/bodyParser/MultiPartBodyParser.swift [hunk 1]
```diff
@@ -21,79 +21,70 @@ class MultiPartBodyParser: BodyParserProtocol {
...
     }
 
     func parse(_ data: NSData) -> ParsedBody? {
         var parts: [Part] = []
-        let bodyLines = divideDataByNewLines(data: data, newLineData: newLineData)
-
-        var bodyLineIterator = bodyLines.makeIterator()
-        skipPreamble(bodyLineIterator: &bodyLineIterator)
-
+        // split the
```
Comment (dfirsht) [inline Sources/Kitura/bodyParser/MultiPartBodyParser.swift]: incomplete comment?

**Comment by dfirsht** [comment]
Hey we are actually moving all development to be PRs against master. Please reopen your PR against the master branch.

**Pull request closed without merging**